packages/vue/src/components/molecules/SfModal/SfModal.js

```diff
@@ -0,0 +1,97 @@
+import SfOverlay from "../../atoms/SfOverlay/SfOverlay.vue";
+
+export default {
+  name: "SfModal",
+  props: {
+    /**
+     * Cross closing modal button
+     */
+    cross: {
+      type: Boolean,
+      default: true
+    },
+    /**
+     * Visibility of the modal
+     */
+    value: {
+      type: Boolean,
+      default: () => false
+    },
+    /**
+     * Whether to show the overlay
+     */
+    overlay: {
+      type: Boolean,
+      default: true
+    },
+    /**
+     * If true clicking outside will not dismiss the modal
+     */
+    persistent: {
+      type: Boolean,
+      default: false
+    },
+    /**
+     * overlay transition effect
+     */
+    transitionOverlay: {
+      type: String,
+      default: "fade"
+    },
+    /**
+     * overlay transition effect
+     */
+    transitionModal: {
+      type: String,
+      default: "fade"
+    }
+  },
+  computed: {
+    visible: {
+      get() {
+        return this.value;
+      },
+      set(val) {
+        this.$emit("input", val);
+      }
+    }
+  },
+  methods: {
+    toggle() {
+      const keydownHandler = e => {
+        if (e.key === "Escape" || e.key === "Esc" || e.keyCode === 27) {
+          this.toggle();
+        }
+      };
+      if (this.visible) {
+        document.addEventListener("keydown", keydownHandler);
+        this.$emit("close");
+      } else {
+        document.removeEventListener("keydown", keydownHandler);
+        this.$emit("open");
+      }
+      this.visible = !this.visible;
+    },
+    checkPersistence() {
+      if (this.persistent === false) {
+        this.toggle();
+      }
+    }
+  },
+  watch: {
+    visible: {
+      handler: value => {
+        if (value && typeof window !== "undefined") {
+          window.document.querySelector("body").style.overflow = "hidden";
```

Comment:
```suggestion
         window.body.style.setProperty("margin-right", `${window.innerWidth - document.body.clientWidth}px`) // better UX
         window.body.style.setProperty("overflow", "hidden")
```
maybe this way is more beauty :) 
what is a better UX? when you add overflow:hidden the scrollbar was remove and content was moved a few pixels (scroll width) when why add margin we doesn't sean this moved, @lorenaramonda wdyt? :)